lib/bundler/cli/plugin.rb

```
@@ -0,0 +1,64 @@
+module Bundler
+  class CLI::Plugin
+    attr_reader :options
+    attr_reader :args
+    def initialize(options, args)
+      @options = options
+      @args = args
+    end
+
+    def run
+      plugins = Bundler.settings.plugins
+      action = @args[0]
+      name = @args[1]
+      if action == "install"
+        begin
+          deps = []
+          deps << Dependency.new("bundler-#{name}", [">= 0"], {})
+          Bundler.set_plugin_install_mode
+          sources = SourceList.new
+          plugin_definition = Definition.new(nil, deps, sources, [])
+          Installer.install_plugin(plugin_definition)
+
+          if !plugins.include?("bundler-#{name}")
+            plugins.push("bundler-#{name}")
+            plugin_definition.specs.to_a.each do |spec|
+              next if spec.name == 'bundler'
+              begin
+                require spec.name
+                Bundler.ui.confirm "Bundler plugin '#{name}' has been successfully installed"
+              rescue LoadError
+                Bundler.ui.error "Bundler plugin '#{name}' couldn't be installed"
+              end
+            end
+          else
+            plugin_definition.specs.to_a.each do |spec|
+              next if spec.name == 'bundler'
+              begin
+                require spec.name
+                Bundler.ui.info "Bundler plugin '#{name}' is already installed"
+              rescue LoadError
+                Bundler.ui.error "Bundler plugin '#{name}' is installed but can't be found in the system"
+              end
+            end
+
+
+          end
+        rescue => e
+          Bundler.ui.error e.message
```

why are we swallowing this error?